assets/context.py

```diff
@@ -0,0 +1,187 @@
+from cogs.utils.embeds import ConfirmEmbed, ConfirmEmbedAnother
+
+argument = namedtuple("Usage", ["required", "text"])
+example = namedtuple("Examples", ["args", "text"])
+sub_desc = namedtuple("subcommands", "desc")
+others = namedtuple("others", ["name", "text"])
+usage_info = namedtuple("usage_information", ["command", "value"])
+keyword = namedtuple("keywords", ["keyword", "desc"])
+
+
+class InvokeError(commands.CommandError):
+    def __init__(self, message):
+        self.message = message
+
+
+class Context(commands.Context):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.InvokeError = InvokeError
+
+    async def confirm(self, title, content, color=misc.random_color()):
+        instance = ConfirmEmbed(self, content, color=color, title=title)
+        confirmation = await instance.start()
+        return confirmation
+
+    async def confirm_another(self, user, title, content, color=misc.random_color()):
+        instance = ConfirmEmbedAnother(ctx=self, user=user, content=content,
+                                       color=color, title=title, dm=True)
+        confirmation = await instance.start()
+        return confirmation
+
+    async def get_response(self, question, check_func=None, delete=True):
+        def validate_response(message):
+            if message.author == self.author and message.channel == self.channel and \
+               True if check_func is None else check_func(message.content):
+                return True
+            else:
+                return False
+
+        question_msg = await self.send(content=question)
+        try:
+            msg = await self.bot.wait_for("message", check=validate_response, timeout=30)
+        except asyncio.TimeoutError:
+            return None
+
+        if delete is True:
+            return msg, question_msg
+        else:
+            return msg
+
+    def error(self, message):
+        raise InvokeError(message)
+
+    async def group_delete(self, *args):
+        try:
+            for msg in args:
+                await msg.delete()
+        except discord.Forbidden:
+            if self.guild:
+                raise commands.BotMissingPermissions("Delete messages!")
+        except (discord.NotFound, discord.HTTPException):
+            pass
+
+    async def send(self, content=None, *, tts=False, embed=None, file=None, files=None, delete_after=None, nonce=None):
+        if self.guild:
+            if not await check.bot_has_ch_perm(self.guild.me, self.channel, send_messages=True):
+                return
+            elif embed and not await check.bot_has_ch_perm(self.guild.me, self.channel, embed_links=True):
+                return await self.send("I don't have permissions to send embeds!")
+
+        return await super().send(content, tts=tts, embed=embed, file=file,
+                                  files=files, delete_after=delete_after,
+                                  nonce=nonce)
+
+
+class Command(commands.Command):
```

Comment:
Knowing that all the help processing should be done in the help command, subclassing `Command` and `Group` becomes useless

Reply:
Knowing that, Imma skip this class for all formatting errors (missing docstrings, lines too longs...)